### ISSUE TITLE
Sweeps - dynamic XCP fee

### ISSUE THREAD

**Issue by jdogresorg** [CounterpartyXCP/counterparty-lib#1220]
Currently sweeps are hardcoded at 0.5 XCP per sweep.

https://github.com/CounterpartyXCP/counterparty-lib/blob/master/counterpartylib/lib/messages/sweep.py#L17-L18

We should change the formula used to calculate the XCP anti-spam fee to be based on the number of balances and issuances being transferred, similar to the way we calculate dynamic XCP anti-spam fee on dividends.

Someone transferring 10,000 asset balances should surely pay a higher anti-spam fee than someone moving just a few hundred asset balances.

Balance transfers should be 1x fee (1 credit, 1 debit)
Asset Transfers should be 2x fee (1 credit, 1 debit, 1 issuance, 1 asset update)

Thanks to @Jpja for pointing out this oversight

**Comment by Jpja** [comment]
I think more than a 2x high fee is needed for asset transfers.

A new row is made in the issuance table which includes the asset description. Descriptions can be very long. The fee should be high enough to prevent a spam attacker to just sweep over and over again an asset with a long description.

With a normal asset transfer, which uses the issuance tx type, the description is included in the tx message and the BTC fee is therefore high with a long description.

Sweep uses a short op_return regardless, so therefore a high XCP fee is needed. Perhaps as high as 0.001 XCP ($0.01 at XCP price $10)? Or ideally, make the fee based on description length? Then it can be a lot cheaper for normal, short-description assets.

**Comment by stet** [comment]
> Here's an example of a very long description, 3600 bytes: https://xchain.io/tx/c1c2be8710eeaeb6976faf0aaea8cdc50363b55ad33caf80e50f8f5fd2864b8c https://blockstream.info/tx/c1c2be8710eeaeb6976faf0aaea8cdc50363b55ad33caf80e50f8f5fd2864b8c
> 
> 8 kB / $103 in fees
> 
> We don't want something like this swept over and over.

Relevant now more than ever. Stamps.

**Comment by Jpja** [comment]
Will this be included in the upcoming 9.60.3 release?

**Comment by jdogresorg** [comment]
Yes, this is on the list of items to be fixed in the 9.60.3 release :)

I think best method going forward is to set a `DB_HIT_COST` cost for a DB hit, which we can easily adjust as price of XCP fluctuates to keep the price "reasonable". 

I propose we initially set `DB_HIT_COST` to 0.001 XCP ($0.003) 

Here is a breakdown of the price logic, hit weight, and example sweeps.

## Current Price Breakdown 
```
XCP Price = $3.00 USD
$3.00 USD / 100,000,000 sats = $0.00000003 USD per XCP sat
0.00100000 XCP = 100,000 XCP sats
100,000 XCP sats x $0.00000003 USD per XCP sat = $0.003

1 DB hit = 0.001 XCP = $0.003
```

## DB Hit weight
```
balances  = 2 db hits
issuances = 4 db hits
```
## Sweep 1 
```
100 balances  = 200 db hits = 0.20 XCP = $0.60
5   issuances =  20 db hits = 0.02 XCP = $0.06
Total XCP Fee = 220 db hits = 0.22 XCP = $0.66
```

## Sweep 2
```
5 balances    =  10 db hits = 0.01 XCP = $0.03
100 issuances = 400 db hits = 0.40 XCP = $1.20
Total XCP Fee = 410 db hits = 0.41 XCP = $1.23
```

## Sweep 3
```
1000 balances  =  2000 db hits = 1.00 XCP = $3.00
1000 issuances =  4000 db hits = 2.00 XCP = $6.00
Total XCP Fee   = 6000 db hits = 3.00 XCP = $9.00
```

## Notes
Asset descriptions stored in the database will be truncated to a max of 1,024 characters going forward once "CIP31 - Enhanced File Encoding Support" is included in release (9.60.3).

Charging XCP fee based on size of description is no longer necessary, since no one will be able to spam more than 1,024 characters into the database via sweeps or issuances after CIP31.

**Comment by Jpja** [comment]
Looks like a good, clean solution.

**Comment by jdogresorg** [comment]
Should be addressed in PR #1268

**Comment by jdogresorg** [comment]
fixed in 9.61.0 release
